test/integration/integration/tests/deploy-api-to-different-listener.go

```diff
@@ -0,0 +1,73 @@
+/*
+ *  Copyright (c) 2023, WSO2 LLC. (http://www.wso2.org) All Rights Reserved.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *
+ */
+
+package tests
+
+import (
+	"testing"
+
+	"github.com/wso2/apk/test/integration/integration/utils/http"
+	"github.com/wso2/apk/test/integration/integration/utils/kubernetes"
+	"github.com/wso2/apk/test/integration/integration/utils/suite"
+)
+
+func init() {
+	IntegrationTests = append(IntegrationTests, APIDifferentListener)
+}
+
+// APIDifferentListener test
+var APIDifferentListener = suite.IntegrationTest{
+	ShortName:   "APIDifferentListener",
+	Description: "An API is deployed to a different listener other than default gateway listener",
+	Manifests:   []string{"tests/api-different-listener.yaml"},
+	Test: func(t *testing.T, suite *suite.IntegrationTestSuite) {
+		ns := "gateway-integration-test-infra"
+		gwAddr := kubernetes.WaitForAPIListenerAddress(t, suite.Client, suite.TimeoutConfig)
```

Comment:
To test the api was added only to the defined listener, we need to invoke both listener addresses, isn't it?
You can call `WaitForGatewayAddress` and get the address with port `9095`, and when invoking it should give a `404`